### PR TITLE
feat: Add CI tests with Redhat UBI

### PR DESCRIPTION
**Pull request by Revolyssup**
### Description

Fixes https://github.com/apache/apisix/issues/9454

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by tao12345666333 — COMMENTED** [review]
Please make CI happy.

File: .github/workflows/redhat-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
+name: CI Redhat UBI - Daily
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
```
Comment (leslie-tsang) [inline .github/workflows/redhat-ci.yaml:5]: no need for schedule job.

Comment (Revolyssup) [reply .github/workflows/redhat-ci.yaml:5]: It was suggested by @tao12345666333.

**Comment by Revolyssup** [comment]
@tao12345666333 Similar to https://openresty.org/package/centos/openresty.repo, is there an rpm package available publicly for redhat? I want to make changes [here](https://github.com/apache/apisix/pull/9552/files#diff-ad9f445050183f2aaf2f188401ee3407cf99a4cb20acb57a78238a40941f63a3R35).

File: ci/redhat-ci.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    install_nodejs
+
+    # install rust
+    install_rust
+
+    # grpc-web server && client
+    cd t/plugin/grpc-web
+    ./setup.sh
+    # back to home directory
+    cd ../../../
```
Comment (leslie-tsang) [inline ci/redhat-ci.sh:82]:
```suggestion
    pushd t/plugin/grpc-web
    ./setup.sh
    # back to home directory
    popd
```

File: ci/redhat-ci.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+#
+
+. ./ci/common.sh
+
+install_dependencies() {
+    export_or_prefix
+
+    # install build & runtime deps
+    yum install -y --disablerepo=* --enablerepo=ubi-8-appstream-rpms --enablerepo=ubi-8-baseos-rpms wget tar gcc automake autoconf libtool make unzip \
+        git sudo openldap-devel which ca-certificates openssl-devel
```
Comment (leslie-tsang) [inline ci/redhat-ci.sh:26]:
```suggestion
    yum install -y --disablerepo=* --enablerepo=ubi-8-appstream-rpms --enablerepo=ubi-8-baseos-rpms \
                   wget tar gcc automake autoconf libtool make unzip git sudo openldap-devel \
                   which ca-certificates openssl-devel 
```

**Comment by Revolyssup** [comment]
@monkeyDluffy6017 Can you rerun failed workflows so that I can confirm that it wasn't a one time issue?

**Comment by Revolyssup** [comment]
@monkeyDluffy6017 Approve CI.

**Comment by Revolyssup** [comment]
@monkeyDluffy6017 The failing tests are unrelated to PR. All tests for the added CI are passing. Can you re-run the CI, I think the tests are flaky.

File: ci/redhat-ci.sh [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    install_curl
+
+    # install openresty to make apisix's rpm test work
+    yum install -y yum-utils && yum-config-manager --add-repo https://openresty.org/package/centos/openresty.repo
+    yum install -y openresty openresty-debug openresty-openssl111-debug-devel pcre pcre-devel xz
+
+
+
+
+
```
Comment (monkeyDluffy6017) [inline ci/redhat-ci.sh:40]: Why so many blanks

Comment (Revolyssup) [reply ci/redhat-ci.sh]: removed extra lines

File: ci/redhat-yum/apache-apisix.repo [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#
```
Comment (soulbird) [inline ci/redhat-yum/apache-apisix.repo:1]: Why we need this?

File: ci/redhat-ci.sh [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    pushd build-cache/
+    # Go is required inside the container.
+    wget -q https://golang.org/dl/go1.17.linux-amd64.tar.gz && tar -xf go1.17.linux-amd64.tar.gz
+    export PATH=$PATH:$(pwd)/go/bin
+    popd
+    # install and start grpc_server_example
+    pushd t/grpc_server_example
+
+    CGO_ENABLED=0 go build
+    ./grpc_server_example \
```
Comment (soulbird) [inline ci/redhat-ci.sh:61]: Look： https://hub.docker.com/r/api7/grpc-server-example/tags

File: ci/redhat-yum/apache-apisix.repo (hunk 5 above)
Comment (Revolyssup) [reply ci/redhat-yum/apache-apisix.repo:1]: In all other files, license started from second line so I left the first line here as well following convention.

Comment (soulbird) [reply ci/redhat-yum/apache-apisix.repo:1]: I mean, why we need this file？

Comment (Revolyssup) [reply ci/redhat-yum/apache-apisix.repo:1]: Yes you're right, we don't need it anymore. In previous commit, I was using it. I removed the file

**Review by soulbird — APPROVED** [review]

**Review by Sn0rt — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**